pkg/cloud/cloud.go

```diff
@@ -868,16 +899,17 @@ func (c *cloud) ResizeDisk(ctx context.Context, volumeID string, newSizeBytes in
 		Size:     aws.Int64(newSizeGiB),
 	}
 
+	klog.Infof("expanding volume %q to size %d", volumeID, newSizeGiB)
 	var mod *ec2.VolumeModification
 	response, err := c.ec2.ModifyVolumeWithContext(ctx, req)
 	if err != nil {
 		if !isAWSErrorIncorrectModification(err) {
 			return 0, fmt.Errorf("could not modify AWS volume %q: %v", volumeID, err)
 		}
 
-		m, err := c.getLatestVolumeModification(ctx, volumeID)
-		if err != nil {
-			return 0, err
+		m, modFetchError := c.getLatestVolumeModification(ctx, volumeID)
```

Comment:
is checking the ModificationState immediately after calling ModifyVolume still necessary, won't waitForVolumeSize do the same thing?

Reply:
I have removed most of this code. Still keeping some of the check in `VolumeModification` object after modification, so as we can return success immediately if volume was previously expnaded to same size already..